openhands/agent_server/docker/Dockerfile

```diff
@@ -56,13 +62,99 @@ COPY --from=ghcr.io/astral-sh/uv /uv /uvx /bin/
+    wget https://github.com/${RELEASE_ORG}/openvscode-server/releases/download/${RELEASE_TAG}/${RELEASE_TAG}-linux-${arch}.tar.gz; \
+    tar -xzf ${RELEASE_TAG}-linux-${arch}.tar.gz; \
+    if [ -d "${OPENVSCODE_SERVER_ROOT}" ]; then rm -rf "${OPENVSCODE_SERVER_ROOT}"; fi; \
+    mv ${RELEASE_TAG}-linux-${arch} ${OPENVSCODE_SERVER_ROOT}; \
+    cp ${OPENVSCODE_SERVER_ROOT}/bin/remote-cli/openvscode-server ${OPENVSCODE_SERVER_ROOT}/bin/remote-cli/code; \
+    rm -f ${RELEASE_TAG}-linux-${arch}.tar.gz; \
+    \
+    # Set proper ownership
+    chown -R ${USERNAME}:${USERNAME} ${OPENVSCODE_SERVER_ROOT}
+
+# Install Docker
+RUN set -eux; \
+    # Determine OS type and install Docker accordingly
+    if grep -q "ubuntu" /etc/os-release; then \
+        # Handle Ubuntu
+        install -m 0755 -d /etc/apt/keyrings; \
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc; \
+        chmod a+r /etc/apt/keyrings/docker.asc; \
+        echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null; \
+    else \
+        # Handle Debian
+        install -m 0755 -d /etc/apt/keyrings; \
+        curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc; \
+        chmod a+r /etc/apt/keyrings/docker.asc; \
+        echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian bookworm stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null; \
+    fi; \
+    # Install Docker Engine, containerd, and Docker Compose
+    apt-get update; \
+    apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin; \
+    apt-get clean; \
+    rm -rf /var/lib/apt/lists/*
+
+# Configure Docker daemon with MTU 1450 to prevent packet fragmentation issues
+RUN mkdir -p /etc/docker && \
+    echo '{"mtu": 1450}' > /etc/docker/daemon.json
+
+# Install Playwright browsers in shared location
+ENV PLAYWRIGHT_BROWSERS_PATH=/opt/playwright-browsers
+RUN set -eux; \
+    mkdir -p /opt/playwright-browsers; \
+    # Install Playwright using uv (since we have it available)
+    uv pip install --system playwright; \
+    playwright install --with-deps chromium; \
+    # Set proper permissions for shared access
+    chmod -R 755 /opt/playwright-browsers; \
+    # Create cache directories and symlinks for both users
+    mkdir -p /home/${USERNAME}/.cache; \
+    mkdir -p /root/.cache; \
+    ln -sf /opt/playwright-browsers /home/${USERNAME}/.cache/ms-playwright; \
+    ln -sf /opt/playwright-browsers /root/.cache/ms-playwright; \
+    chown -h ${USERNAME}:${USERNAME} /home/${USERNAME}/.cache/ms-playwright; \
+    # Set environment variable for all users
+    echo 'export PLAYWRIGHT_BROWSERS_PATH=/opt/playwright-browsers' >> /etc/environment
+
```

Comment:
Let's hold off playwright for now -- i want to test it in a separate PR once we figured out this: https://github.com/All-Hands-AI/agent-sdk/issues/398#issuecomment-3318195932

```suggestion
```